src/uwtools/tests/test_rocoto.py

```diff
@@ -129,6 +129,70 @@ def test__add_task_dependency_fail(self, instance, root):
         with raises(UWConfigError):
             instance._add_task_dependency(e=root, config=config)
 
+    def test__add_task_dependency_and(self, instance, root):
+        config = {"and": {"or": {"datadep": {"attrs": {"age": "120"}}}}}
+        instance._add_task_dependency(e=root, config=config)
+        dependency = root[0]
+        assert dependency.tag == "dependency"
+        and_ = dependency[0]
+        assert and_.tag == "and"
+        # assert and_.get("age") == "120"
```

Comment:
Still working out this test case. With the updates I made to the methods, I believe I have to do a similar thing to this test as I did to `test__add_task_dependency_operand`, where I need to loop over `for attr, val in block["attrs"].items():`. I currently get `None` back as the error message for `and_.get("age")`

Reply:
I haven't been able to figure this one out. Since `attrs` is nested deeper into the config I attempted to use `next(iter(config))` but it began to look quite messy.

Reply:
I'll fetch your branch and take a look.

Reply:
All comments should be addressed now and this is the final outstanding issue.

Reply:
`age` is not an attribute of `and`, it's an attribute of `datadep`, which is two levels deeper:
```
(Pdb) ll
132  	    def test__add_task_dependency_and(self, instance, root):
133  	        config = {"and": {"or_get_obs": {"datadep": {"attrs": {"age": "120"}}}}}
134  	        breakpoint()
135  	        instance._add_task_dependency(e=root, config=config)
136  	        dependency = root[0]
137  	        assert dependency.tag == "dependency"
138  	        and_ = dependency[0]
139  	        assert and_.tag == "and"
140  ->	        assert and_.get("age") == "120"
(Pdb) p and_.getchildren()[0].getchildren()[0].get("age") == "120"
True
```

Reply:
Interesting use of the `getchildren()` attr on the `and_` tag. I was doing something similar with:

``` next(iter(and_))[0]```

I'll try this instead for my test case. It'll also be easier to breakdown in the long run vs. nested `next(iter(...))` which were looking messy.

Reply:
If nothing else, `.getchildren()` might make the intent a little clearer.